File: challenge-374/abigail/perl/ch-2.pl
#!/opt/perl/bin/perl

use 5.042;

use strict;
use warnings;
no  warnings 'syntax';

while (my $line = <>) {
    my $max = 0;
    while ($line =~ /([1-9])\1*/gp) {
        #
        # Iterate over all the sequences of the same digit, 
        # remember to largest (in a numerical sense) seen.
        #
        $max = ${^MATCH} if ${^MATCH} > $max
    }
    say $max;
}

__END__

File: challenge-374/abigail/ruby/ch-2.rb
#!/opt/homebrew/opt/ruby/bin/ruby

ARGF . each_line do |line|
    max = 0
    line . scan /([1-9])\1*/ do             # Iterate over all matches
        max = $& . to_i if $& . to_i > max  # Remember the largest match
    end
    puts max
end
